core/src/getCollateralPriceOracleConfig.ts

```diff
@@ -0,0 +1,210 @@
+    const contractOracleMap = new ethers.Contract(
+        contractOracleMapAdderss,
+        contractCollateralToOracleInterface,
+        signer
+    );
+    const address: string = (await contractOracleMap.ilks(ethers.utils.formatBytes32String(collateralType))).pip;
+    const contract = new ethers.Contract(address, contractOracleInterface, signer);
+    return { contract, address };
+};
+const overwriteValue = async (contractAddress: string, newValue: BigNumber, slot: number) => {
+    overwriteUintValueInAddress(contractAddress, ethers.utils.hexlify(slot), newValue);
+};
+const callContractFunctionOrThrow = async (contract: ethers.Contract, functionName: string, ...args: any[]) => {
+    console.log(...args);
+    return await contract[functionName](...args);
+};
+const callFunction = async (contract: ethers.Contract, functionName: string, ...args: any[]): Promise<string> => {
+    const returnedValueHex = await callContractFunctionOrThrow(contract, functionName, ...args);
+    return returnedValueHex._hex;
+};
+const addToWhitelist = async (contractAddress: string, whitelistSlot: number) => {
+    const slotAddress = generateMappingSlotAddress(`0x${whitelistSlot.toString(16)}`, HARDHAT_PUBLIC_KEY);
+    await overwriteUintValueInAddress(contractAddress, slotAddress, new BigNumber(1));
+};
+const runOverwriteStep = async (
+    contract: ethers.Contract,
+    address: string,
+    functionName: string,
+    offset?: number,
+    length?: number
+) => {
+    const previousValueRaw = await callFunction(contract, functionName);
+    const previousValue = new BigNumber(
+        offset && length ? previousValueRaw.substring(offset, offset + length) : previousValueRaw
+    );
+    const { slot, newValue } = await promptSlotNumberAndNewValue();
+    await overwriteValue(address, new BigNumber(newValue), slot);
+    const currentPriceRaw = await callFunction(contract, functionName);
+    const currentPrice = new BigNumber(
+        offset && length ? currentPriceRaw.substring(offset, offset + length) : currentPriceRaw
+    );
+    console.info(`Value returned before overwrite: ${previousValue}, Value returned after overwrite: ${currentPrice}`);
+};
+const run = async () => {
+    await resetNetworkAndSetupWallet(undefined, HARDHAT_PRIVATE_KEY);
+    const basicInfo = await promptBasicInformation();
+    const collateralType = await promptCollateralType();
+    const { contract, address } = await getOracleAddressAndContract(collateralType);
+    console.info(`Contract address: ${address}`);
+    if (basicInfo.hasWhitelist) {
+        const { whitelistSlot, whitelistFunction } = await promptWhitelistSlotAndFunction();
+        await addToWhitelist(address, parseInt(whitelistSlot));
+        const isWhitelisted = await callFunction(contract, whitelistFunction, HARDHAT_PUBLIC_KEY);
+        if (isWhitelisted === '0x00') {
+            throw new Error('Failed to whitelist the wallet on the fork');
+        }
```

Comment:
@valiafetisov here the logic should overwrite the mapping, but it doesn't do it AND does not throw an error.
 - does not seem that it's the issue with address calculation because
   - calculating the address of the whitelisting flag for `spot` contract instead of hardhat public key provides the expected/correct result (flag in the auth mapping is `1`)
   - if i set the value by `hardhat_setStorageAt` to 1 and read it by `eth_getStorageAt`, i still get 0 while keeping track of the same address being used.

Reply:
If you will want to run it yourself, here's the current input log to reach the error:

```
> npx hardhat run ./src/getCollateralPriceOracleConfig.ts                                                                                                         Y liquidation_core_functionality 21.Sep.22__12:44 ‚ôøÔ∏è
Forking at the latest block
New RPC endpoint has started on http://localhost:8545
Using wallet with:
         - public key: 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
         - private key: 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
- Does the contract have whitelisting for method execution? : yes
- Does contract provide future price? : yes
- State the collateral type. E.g. ETH-A : ETH-A
Contract address: 0x81FE72B5A8d1A857d176C3E7d5Bd2679A9B85763

    For this step you must guess the slot number.
    1. Find the function that returns the variable, provide its name.
    2. Find the variable you're interested in the the smart contract.
    3. Count variables from top to bottom - what's the number of the variable you're interested in? Provide it.
    4. If the variable is less than 32B (e.g. uint256) provide the offset and length to indicate the start of the variable.

    Example for point (4):
        boolean takes up less than 32B of memory, the let's assume the returned value from slot address looks like '0x1000000000000000000000000000000000000000000000000000000000000000'.
        The offset is 3, length is 1.


- What is the slot of the whitelist mapping?: 5
- What is the function to check the whitelist?: bud
0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266

/unified-auctions-ui/core/src/getCollateralPriceOracleConfig.ts:193
            throw new Error('Failed to whitelist the wallet on the fork');
                  ^
Error: Failed to whitelist the wallet on the fork
    at run (/unified-auctions-ui/core/src/getCollateralPriceOracleConfig.ts:193:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

Reply:
Aaaah, great job rubber-ducking, @valiafetisov , i knew you're the best helper here.

`npx hardhat --network testnetwork --config local.hardhat.config.ts run ./src/getCollateralPriceOracleConfig.ts` instead of `npx hardhat run ./src/getCollateralPriceOracleConfig.` fixes it